ui-tests/src/test/java/io/syndesis/qe/fragments/common/form/CoupledForm.java

```diff
@@ -0,0 +1,113 @@
+package io.syndesis.qe.fragments.common.form;
+
+import static com.codeborne.selenide.Selenide.$;
+
+import org.openqa.selenium.By;
+
+import com.codeborne.selenide.SelenideElement;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class CoupledForm extends Form{
+
+    private String[] selectValues = new String[] {"yes","checked","check","select", "selected"};
+    private String[] unselectValues = new String[] {"no","unchecked","uncheck","unselect", "unselected"};
+
+    public CoupledForm(SelenideElement rootElement) {
+        super(rootElement);
+    }
+
+    @Override
+    public void fill(Map<String, String> data) {
+        if (data.isEmpty()) {
+            throw new IllegalArgumentException("There are no data to be filled into a form.");
+        }
+
+        for(String label: data.keySet()) {
+            log.info("Filling form: " + label + " | " + data.get(label));
```

Comment:
I'd avoid logging values from input map as it might contain sensitive info like password, access token, etc.

Reply:
ok